pkg/provider/azure_vmss_cache.go

```diff
@@ -415,9 +415,20 @@ func (ss *ScaleSet) getVMManagementTypeByNodeName(nodeName string, crt azcache.A
 	}
 
 	cachedNodes := cached.(NonVmssUniformNodesEntry).ClusterNodeNames
-
 	// if the node is not in the cache, assume the node has joined after the last cache refresh and attempt to refresh the cache.
 	if !cachedNodes.Has(nodeName) {
+		if cached.(NonVmssUniformNodesEntry).AvSetVMNodeNames.Has(nodeName) {
```

Comment:
Why do we want to check it twice?

Reply:
this is for checking nodeName in cache: `cached.(NonVmssUniformNodesEntry).AvSetVMNodeNames`, and in csi driver, L419 `cachedNodes` is always empty.